### PR TITLE
Break search for let identifiers on var decl

### LINKED ISSUE #1671: Transpilation problem with a global, shadowed let, a for-of loop and an arrow function in conjunction
Compiling the program

```javascript
let c;

function f(c) {
  var p;
  for (p of g(() => c));
}
```

yields a

    RangeError: Maximum call stack size exceeded.

The error does not occur if `c` is not declared at the global level or if the argument of `f` does not shadow it. If `let` is substituted by `var` on the global level, the error also does not occur. Nor does it occur if I leave out the `for` construct so that only `g(() => c)` remains. Leaving out `g` still produces the error.

### PR DESCRIPTION
**Pull request by johnjbarton**
Fixes #1671

### CODE REVIEW

**Comment by johnjbarton** [comment]
I would have more confidence if we could come up with other cases that fail.

File: src/codegeneration/BlockBindingTransformer.js [hunk 1]
```diff
@@ -766,6 +766,9 @@ class FindBlockBindingInLoop extends FindVisitor {
             if (fnScope.hasLexicalBindingName(identifierToken)) {
               return true;
             }
+            if (fnScope.hasVariableBindingName(identifierToken)) {
```
Comment (arv) [inline src/codegeneration/BlockBindingTransformer.js]: Change this to hasBinding which handles both lexical and variable.

**Comment by arv** [comment]
LGTM - It fixes the bug and does not break any of the existing tests...

But like you I feel bad about this. I'm not confident this works correctly. This code is too gnarly for me to understand. At some point I should go back and see if we can come up with a more maintainable block binding transformer.

**Comment by arv** [comment]
@rodyhaddad Care to take a look as well?

**Pull request closed without merging**